javaparser-core-generators/src/main/java/com/github/javaparser/generator/VisitorGenerator.java

```diff
@@ -99,5 +103,18 @@ private void generateVisitMethodForNode(BaseNodeMetaModel node, ClassOrInterface
         }
     }
 
+    /**
+     * Get the list of properties in the order they should be visited.
+     *
+     * @param baseNodeMetaModel The metamodel to get the properties.
+     *
+     * @return The list of properties.
+     */
+    public List<PropertyMetaModel> getAllPropertyMetaModels(BaseNodeMetaModel baseNodeMetaModel) {
+        List<PropertyMetaModel> propertiesList = new ArrayList<>(baseNodeMetaModel.getAllPropertyMetaModels());
+        Collections.reverse(propertiesList);
+        return propertiesList;
+    }
```

Comment:
This is the only place that I could find which does any sorting (in this case, reversing) of the properties. Is this where the deterministic order happens?